### PR TITLE
(docs): update README.md

### PR DESCRIPTION
**Pull request by vladdoster**
- correct capitalization
- correct grammar
- add punctuation
- increase readability
- remove extra blank lines
- add language hint to code fences

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ On OS X, the latest release of Libfabric can be installed using the
 [Homebrew](https://github.com/Homebrew/homebrew) package manager using the
 following command:
 
-```
-$ brew install libfabric
+```bash
+brew install libfabric
```
Comment (j-xiong) [inline README.md:22]: The convention is to put '$' before the shell command line. No need to change that unless there is a strong reason to do so. This applies to other similar changes in this commit.

File: README.md [hunk 2]
```diff
@@ -213,23 +212,22 @@ See the `fi_sockets(7)` man page for more details.
 
 ***
 
-The tcp provider is an optimized socket based provider that supports
+The TCP provider is an optimized socket-based provider that supports
```
Comment (j-xiong) [inline README.md:215]: tcp here is the provider name, not the protocol. can change to `tcp` instead.

File: README.md [hunk 3]
```diff
@@ -213,23 +212,22 @@ See the `fi_sockets(7)` man page for more details.
...
 reliable connected endpoints.  It is intended to be used directly by
 apps that need MSG endpoint support, or in conjunction with the rxm
-provider for apps that need RDM endpoints.  The tcp provider targets
+provider for apps that need RDM endpoints.  The TCP provider targets
```
Comment (j-xiong) [inline README.md:218]: `tcp`

**Comment by shefty** [comment]
Manually pulled in most proposed updates, plus added a few others, and added to PR #7278.  Thanks!

File: README.md [hunk 4]
```diff
@@ -95,17 +95,17 @@ This enables or disables the provider named `<provider>`. Valid options are:
 
 Consider the following example:
 
-```
+```bash
 $ ./configure --prefix=/opt/libfabric --disable-sockets && make -j 32 && sudo make install
```
Comment (acgoldma) [inline README.md:99]: You removed `$` all but here.

Comment (shefty) [reply README.md:99]: These were the original changes, which were not merged.  The merged changes keep the $ in all locations.

**Pull request closed without merging**